[v8.6.0] Create Helm Chart from Static Manifests

hswong3i opened the pull request:
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR convert our static deployment manifests from `deploy/kubernetes/*` into Helm Charts under `charts/*`, by running a convert script under `deploy/util/chart-releaser.sh`.

Moreover, it handle the chart release with branch `gh-pages`, by integrate with https://github.com/helm/chart-releaser-action.

The sample helm repo could find from https://alvistack.github.io/kubernetes-csi-external-snapshotter/index.yaml, by:

    helm repo add external-snapshotter https://alvistack.github.io/kubernetes-csi-external-snapshotter
    helm repo update
    helm search repo external-snapshotter

Before each stable tag release, maintainers only required to:
1. Run the `./deploy/util/chart-releaser.sh` and update templates under `charts/*/templates/*`
2. Double confirm `charts/*/values.yml` with correct values, e.g. image tags
3. `git add --all --force charts/* && git commit`

**Which issue(s) this PR fixes**:
Fixes #551
Fixes #622
Fixes #751

```release-note
Create Helm Chart from Static Manifests
```

hswong3i commented:
Sorry @TerryHowe may we continuous the review for https://github.com/kubernetes-csi/external-snapshotter/pull/1285 with this new PR?

hswong3i commented on deploy/util/chart-releaser.sh:
```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bash
+
+# Copyright 2025 The Kubernetes Authors.
```
> Added in another PR

Which PR?

TerryHowe replied:
I think it is generally preferred to have one change to one PR and this change is not related.

It is like going to the store to buy mangos and you have to take a pound of hamburger with it.

hswong3i replied:
Please refer to my original statement:

> Before each stable tag release, maintainers only required to:
> 
> - Run the ./deploy/util/chart-releaser.sh and update templates under charts/*/templates/
> - Double confirm charts/*/values.yml with correct values, e.g. image tags
> - git add --all --force charts/* && git commit

This PR is target to simplify workload for our developers, where the Chart could automatically release with GitHub Action. 

Without the changes to `deploy/util/chart-releaser.sh`, this PR will become half made...

humblec commented:
cc @xing-yang 
Definitely I see this add value to the project

hswong3i commented:
Seems like our CI officially failing?

humblec committed:
deploy: update sidecar image versions

Update cross-referenced sidecar images to latest stable releases:
- csi-provisioner: v6.1.0 -> v6.2.0
- csi-snapshotter: v8.4.0 -> v8.5.0
- snapshot-controller: v8.4.0 -> v8.5.0
- hostpathplugin: v1.11.0 -> v1.17.1

Note: snapshot-conversion-webhook left at v8.0.1 as no published
images exist in registry.k8s.io for this component.

Signed-off-by: Humble Devassy Chirammal <humble.devassy@gmail.com>

hswong3i committed:
Create Helm Chart from Static Manifests

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR convert our static deployment manifests from `deploy/kubernetes/*` into Helm Charts under `charts/*`, by running a convert script under `deploy/util/chart-releaser.sh`.

Moreover, it handle the chart release with branch `gh-pages`, by integrate with https://github.com/helm/chart-releaser-action.

The sample helm repo could find from https://alvistack.github.io/kubernetes-csi-external-snapshotter/index.yaml, by:

    helm repo add external-snapshotter https://alvistack.github.io/kubernetes-csi-external-snapshotter
    helm repo update
    helm search repo external-snapshotter

Before each stable tag release, maintainers only required to:
1. Run the `./deploy/util/chart-releaser.sh` and update templates under `charts/*/templates/*`
2. Double confirm `charts/*/values.yml` with correct values, e.g. image tags
3. `git add --all --force charts/* && git commit`

Signed-off-by: Wong Hoi Sing Edison <hswong3i@pantarei-design.com>
